src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapGet_NoParam_StringReturn_WithFilter.generated.txt

```diff
@@ -0,0 +1,206 @@
+#nullable enable
+using global::Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    %GENERATEDCODEATTRIBUTE%
+    internal class SourceKey
+    {
+        public string Path { get; init; }
+        public int Line { get; init; }
+
+        public SourceKey(string path, int line)
+        {
+            Path = path;
+            Line = line;
+        }
+    }
+}
+
+// This class needs to be internal so that the compiled application
+// has access to the strongly-typed endpoint definitions that are
+// generated by the compiler so that they will be favored by
+// overload resolution and opt the runtime in to the code generated
+// implementation produced here.
+%GENERATEDCODEATTRIBUTE%
+internal static class GenerateRouteBuilderEndpoints
+{
+    private static readonly string[] GetVerb = new[] { HttpMethods.Get };
+    private static readonly string[] PostVerb = new[] { HttpMethods.Post };
+    private static readonly string[] PutVerb = new[]  { HttpMethods.Put };
+    private static readonly string[] DeleteVerb = new[] { HttpMethods.Delete };
+    private static readonly string[] PatchVerb = new[] { HttpMethods.Patch };
+
+    %GENERATEDCODEATTRIBUTE%
+internal static global::Microsoft.AspNetCore.Builder.RouteHandlerBuilder MapGet(
+        this global::Microsoft.AspNetCore.Routing.IEndpointRouteBuilder endpoints,
+        [global::System.Diagnostics.CodeAnalysis.StringSyntax("Route")] string pattern,
+        global::System.Func<string> handler,
+        [global::System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [global::System.Runtime.CompilerServices.CallerLineNumber]int lineNumber = 0)
+    {
+        return global::Microsoft.AspNetCore.Http.Generated.GeneratedRouteBuilderExtensionsCore.MapCore(endpoints, pattern, handler, GetVerb, filePath, lineNumber);
+    }
+
+}
+
+namespace Microsoft.AspNetCore.Http.Generated
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using System.IO;
+    using Microsoft.AspNetCore.Routing;
+    using Microsoft.AspNetCore.Routing.Patterns;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.FileProviders;
+    using Microsoft.Extensions.Primitives;
+
+    using MetadataPopulator = System.Func<System.Reflection.MethodInfo, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions?, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult>;
+    using RequestDelegateFactoryFunc = System.Func<System.Delegate, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult?, Microsoft.AspNetCore.Http.RequestDelegateResult>;
```

Comment:
These can be simplified now, for readability.

```suggestion
    using MetadataPopulator = Func<MethodInfo, RequestDelegateFactoryOptions?, RequestDelegateMetadataResult>;
    using RequestDelegateFactoryFunc = Func<Delegate, RequestDelegateFactoryOptions, RequestDelegateMetadataResult?, RequestDelegateResult>;
```

Reply:
Not sure it can be. Last I tried this doesn't work, or the IDE was trolling me.

Reply:
I'd be really surprised if this doesn't work. Note that @captainsafia refactored the generated code to be inside a namespace with all the required usings inside the namespace declaration. With that work, this code can be written just like a "normal dev" would write code.

Reply:
It looks like fully-qualified namespaces are a requirement of type aliases regardless of what other usings exist ([ref](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive#using-alias)).

> Create a using alias directive to make it easier to qualify an identifier to a namespace or type. In any using directive, the fully qualified namespace or type must be used regardless of the using directives that come before it.